protocol-units/da-sequencer/node/src/celestia/mod.rs

```diff
@@ -241,58 +261,74 @@ impl<B: BlockOps, C: CelestiaClientOps> DaSequencerExternalDa for CelestiaExtern
 #[cfg(test)]
 mod tests {
 	use super::*;
-	use movement_types::block::{Block, BlockMetadata, Id};
+	use movement_types::block::{self, Block, BlockMetadata};
 	use movement_types::transaction::Transaction;
 	use rand::Rng;
 	use std::collections::{BTreeSet, HashMap};
 	use std::sync::Arc;
 	use tokio::sync::RwLock;
 
-	fn digest(block: &Block, block_height: BlockHeight) -> SequencerBlockDigest {
-		SequencerBlockDigest::new(block_height, Id::new(*block.id().as_bytes()))
-	}
-
-	fn into_digests(blocks: &[Block]) -> Vec<SequencerBlockDigest> {
-		blocks
-			.iter()
-			.enumerate()
-			.map(|(height, block)| digest(block, BlockHeight::from(height as u64)))
-			.collect()
+	fn into_ids(blocks: &[Block]) -> Vec<block::Id> {
+		blocks.iter().map(|block| block.id()).collect()
 	}
 
 	#[derive(Clone, Default, Debug, Eq, PartialEq)]
 	enum CelestiaClientCalls {
 		#[default]
 		Noop,
-		SendBlock(SequencerBlockDigest, BlockSource),
+		SendBlock(block::Id, BlockSource),
 		GetBlobsAtHeight(CelestiaHeight),
 	}
 
 	#[derive(Clone, Default, Debug, Eq, PartialEq)]
 	enum BlockProviderCalls {
 		#[default]
 		Noop,
-		NotifyBlockCommited(BlockHeight, CelestiaHeight),
-		RequestBlock(BlockHeight),
+		NotifyBlocksCommitted(Vec<block::Id>, CelestiaHeight),
+		RequestBlockAtHeight(BlockHeight),
+		RequestBlockForId(block::Id),
+	}
+
+	trait Id {
+		fn id(&self) -> block::Id;
+	}
+
+	impl Id for Block {
+		fn id(&self) -> block::Id {
+			self.id()
+		}
+	}
+
+	impl Id for CelestiaBlob {
+		fn id(&self) -> block::Id {
+			self.iter().next().copied().unwrap()
+		}
```

Comment:
The generalization seems fishy, but this is test code.

Reply:
Haha, right, this is nonsense. But it's not used in any tests. I needed `Id` for `Blocks` to index them by height and id. There is no feature to get a blob by ID.